basic_auth/api/application.py

```diff
@@ -41,8 +45,22 @@ def __init__(self, name, resource):
             self._collection_methods_map)
         payload = await self._validate_request(request, allowed_methods)
 
+        date_format = "%Y-%m-%d-%H-%M"
+
+        start_date = request.query.get('start_date')
+        if start_date is not None:
+            start_date = datetime.strptime(start_date, date_format)
+
+        end_date = request.query.get('end_date')
+        if end_date is not None:
+            end_date = datetime.strptime(end_date, date_format)
+
         func = getattr(
             self.resource, self._collection_methods_map[request.method])
+
+        if request.method == 'GET':
```

Comment:
So parsing dates could be done inside this if block, given that dates seems not to be used otherwise.